File: Spring/Chapter6/JavaDemo/MyBaitsSpringDemo/src/main/java/pers/dao/UserDao.java
package pers.dao;

import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Repository;
import pers.entity.User;

import java.util.List;

@Repository
@Mapper
public interface UserDao {
    User selectById(Integer id);
    List<User> selectAll();
    int insert(User user);
    int update(User user);
    int delete(Integer id);
}

File: Spring/Chapter6/KotlinDemo/MyBaitsSpringDemo/src/main/kotlin/pers/dao/UserDao.kt
package pers.dao

import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Repository
import pers.entity.User

@Mapper
@Repository
interface UserDao {
    fun selectById(id:Int):User
    fun selectAll():List<User>
    fun insert(user: User):Int
    fun update(user: User):Int
    fun delete(id: Int):Int
}